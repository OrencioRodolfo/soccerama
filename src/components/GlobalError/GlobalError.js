import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import Snackbar from '../Snackbar/Snackbar'
import actions from '../../store/actions'

const GlobalError = ({ errors, dismissError }) => errors.map((error, index) => (
  <Snackbar
    key={`${error.code}-${index}`} // eslint-disable-line
    open={!error.read}
    message={error.message}
    onClose={() => { dismissError(index) }}
  />
))

GlobalError.propTypes = {
  errors: PropTypes.arrayOf(PropTypes.shape({
    code: PropTypes.number,
    message: PropTypes.string.isRequired,
  })).isRequired,
  dismissError: PropTypes.func,
}

GlobalError.defaultProps = {
  dismissError: () => {},
}

const mapStateToProps = ({ api }) => ({
  errors: api.errors,
})

export default connect(mapStateToProps, {
  dismissError: actions.dismissError,
})(GlobalError)

export {
  GlobalError as DGlobalError,
  mapStateToProps,
}
