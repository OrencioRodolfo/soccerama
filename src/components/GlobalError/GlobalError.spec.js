import React from 'react'
import { shallow } from 'enzyme'
import { DGlobalError, mapStateToProps } from './GlobalError'

describe('Global error component', () => {
  let errors

  beforeEach(() => {
    errors = [{
      code: 1,
      message: 'sample',
    }]
  })

  it('must render properly', () => {
    const component = shallow(<DGlobalError errors={errors} />)
    expect(component).toMatchSnapshot()
  })

  describe('mapStateToProps function', () => {
    it('must return the api errors', () => {
      const result = mapStateToProps({
        api: {
          errors: [],
        },
      })
      expect(result).toEqual({ errors: [] })
    })
  })
})
