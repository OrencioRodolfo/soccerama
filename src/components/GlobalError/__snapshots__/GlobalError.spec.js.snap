// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Global error component must render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GlobalError
    errors={
      Array [
        Object {
          "code": 1,
          "message": "sample",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Array [
    <WithStyles(CustomSnackbar)
      message="sample"
      onClose={[Function]}
      open={true}
    />,
  ],
  Symbol(enzyme.__nodes__): Array [
    Array [
      <WithStyles(CustomSnackbar)
        message="sample"
        onClose={[Function]}
        open={true}
      />,
    ],
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
